src/main/java/org/testng/TestNG.java

```diff
@@ -137,7 +137,7 @@
   private List<IClassListener> m_classListeners = Lists.newArrayList();
   private List<ITestListener> m_testListeners = Lists.newArrayList();
   private List<ISuiteListener> m_suiteListeners = Lists.newArrayList();
-  private Set<IReporter> m_reporters = Sets.newHashSet();
+  private Map<Class< ? extends IReporter>,IReporter> m_reporters = Maps.newHashMap();
```

Comment:
I am guessing you meant `bad formatting` when you said `typo`. Correct ? Fixed the formatting.